app/src/NativeBase/Components/Forms/DatePicker.tsx

```diff
@@ -31,10 +31,14 @@ const DatePicker: FC<DatePickerProps> = ({
   const { colorMode } = useColorMode()
   const { width } = Dimensions.get('window')
 
+  const showDatePicker = () => {
+    setPickerActive(!pickerActive)
+  }
+
   const handleChange = (value: Date) => {
     onChange(value)
-    setSelectedStartDate(value)
-    setPickerActive(false)
+    setSelectedStartDate(selectedStartDate)
```

Comment:
This needs to stay as `setSelectedStartDate(value)`, otherwise we're just setting `selectedStartDate` to the value it already is, which means it always stays as `null`.  Visually if it's `null` then the date the user's chosen isn't highlighted on the calendar if they reopen it.

Was `selectedStartDate` causing the bug in iOS?

Reply:
Oh yes, that was my bad - I was tweaking the function as I suspected it might have not been updating the state correctly. The culprit was Pressable component and the fact it was **not** pressable - it needed surrounded with a box/view to enable for the state changes and calendar to show up